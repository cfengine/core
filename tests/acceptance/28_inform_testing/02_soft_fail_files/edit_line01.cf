body file control
{
  inputs => { "$(sys.policy_entry_dirname)/../common_soft_fail.cf.sub" };
}

bundle common testcase
{
  vars:
    "filename" string => "$(this.promise_filename)";
}


bundle agent setup
{
  files:
    "$(G.testfile)"
      edit_line => insert_lines("
foo 1
bar 2
baz 3
me:x:1001:1001::/home/me:/bin/sh
"),
      create => "true";
}

bundle agent main
{
  files:
    "$(G.testfile)"
      edit_line => set_colon_field("me","6","/my/new/shell");
}

bundle agent teardown
{
  reports:
    "Edited file has contents: "
      printfile => my_cat( $(G.testfile) );
}

body printfile my_cat(file)
# @brief Report the contents of a file
# @param file The full path of the file to report
{
        file_to_print => "$(file)";
        number_of_lines => "inf";
}

