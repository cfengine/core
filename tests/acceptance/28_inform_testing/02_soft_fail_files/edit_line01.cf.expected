    info: files promise '/tmp/TEST.cfengine' repaired, edit_line => set_colon_field("me","6","/my/new/shell");
    info: Set field 6 with separator ':' to value '/my/new/shell' on line 4 of '/tmp/TEST.cfengine'
